manticore/core/manticore.py

```diff
@@ -227,66 +137,14 @@ def unregister_plugin(self, plugin):
         plugin.manticore = None
 
     def __del__(self):
-        plugins = list(self.plugins)
+        # If an exception is thrown in a child/inherited class's __init__ before calling super().__init__
+        # the self.plugins is not assigned
+        # because of that, we need to use a `getattr` here, otherwise we might get two exceptions
+        # (and the one from here is irrelevant and confusing)
+        plugins = list(getattr(self, 'plugins', []))
         for plugin in plugins:
             self.unregister_plugin(plugin)
 
-    @classmethod
-    def linux(cls, path, argv=None, envp=None, entry_symbol=None, symbolic_files=None, concrete_start='', stdin_size=STDIN_INPUT_DEFAULT_SIZE, **kwargs):
-        """
-        Constructor for Linux binary analysis.
-
-        :param str path: Path to binary to analyze
-        :param argv: Arguments to provide to the binary
-        :type argv: list[str]
-        :param envp: Environment to provide to the binary
-        :type envp: dict[str, str]
-        :param entry_symbol: Entry symbol to resolve to start execution
-        :type envp: str
-        :param symbolic_files: Filenames to mark as having symbolic input
-        :type symbolic_files: list[str]
-        :param str concrete_start: Concrete stdin to use before symbolic input
-        :param int stdin_size: symbolic stdin size to use
-        :param kwargs: Forwarded to the Manticore constructor
-        :return: Manticore instance, initialized with a Linux State
-        :rtype: Manticore
-        """
-        try:
-            return cls(make_linux(path, argv, envp, entry_symbol, symbolic_files, concrete_start, stdin_size), **kwargs)
-        except elftools.common.exceptions.ELFError:
-            raise Exception(f'Invalid binary: {path}')
-
-    @classmethod
-    def decree(cls, path, concrete_start='', **kwargs):
-        """
-        Constructor for Decree binary analysis.
-
-        :param str path: Path to binary to analyze
-        :param str concrete_start: Concrete stdin to use before symbolic input
-        :param kwargs: Forwarded to the Manticore constructor
-        :return: Manticore instance, initialized with a Decree State
-        :rtype: Manticore
-        """
-        try:
-            return cls(make_decree(path, concrete_start), **kwargs)
-        except KeyError:  # FIXME(mark) magic parsing for DECREE should raise better error
-            raise Exception(f'Invalid binary: {path}')
-
-    @classmethod
-    def evm(cls, **kwargs):
```

Comment:
**I removed this** instead of moving to `ethereum/manticore.py:Manticore`. I feel like this is not used.
@feliam @mossberg @yan please reply if it is not okay/if we use this classmethod anywhere.

Reply:
This is likely okay, since we're not after native/evm symmetry anyway.